zkevm-circuits/src/state_circuit/state.rs

```diff
@@ -98,453 +75,327 @@ pub struct Config<
 
-        let address_diff_is_zero = IsZeroChip::configure(
-            meta,
-            |meta| {
-                let padding = meta.query_advice(padding, Rotation::cur());
-                let is_not_padding = one.clone() - padding;
-                let q_target = meta.query_fixed(q_target, Rotation::cur());
-                let q_not_first = q_target.clone() * (q_target - one.clone());
-
-                q_not_first * is_not_padding
-            },
-            |meta| {
-                let address_cur = meta.query_advice(address, Rotation::cur());
-                let address_prev = meta.query_advice(address, Rotation::prev());
-                address_cur - address_prev
-            },
-            address_diff_inv,
-        );
-
-        // Only one monotone gadget is used for memory and stack (with
-        // MEMORY_ADDRESS_MAX as it is bigger)
-        let address_monotone = MonotoneChip::<F, MEMORY_ADDRESS_MAX, true, false>::configure(
-            meta,
-            |meta| {
-                let padding = meta.query_advice(padding, Rotation::cur());
-                let is_not_padding = one.clone() - padding;
-                // Since q_memory_not_first and q_stack_non_first are
-                // mutually exclusive, q_not_first is binary.
-                let q_not_first = q_memory_not_first(meta) + q_stack_not_first(meta);
-
-                q_not_first * is_not_padding
-            },
-            address,
-        );
-
-        // Padding monotonicity could be checked using gates (as padding only
-        // takes values 0 and 1), but it's much slower than using a
-        // lookup.
-        let padding_monotone = MonotoneChip::<F, 1, true, false>::configure(
-            meta,
-            |meta| q_memory_not_first(meta) + q_stack_not_first(meta),
-            padding,
-        );
-
-        // A gate for the first row (does not need Rotation::prev).
-        meta.create_gate("First memory row operation", |meta| {
-            let value = meta.query_advice(value, Rotation::cur());
-            let flag = meta.query_advice(flag, Rotation::cur());
-            let q_read = one.clone() - flag;
-            let q_memory_first = q_memory_first(meta);
-
-            // read value must be 0
-            vec![q_memory_first * q_read * value]
-        });
-
-        meta.create_gate("Memory operation + padding", |meta| {
-            // if is_read:
-            //      if address_cur == address_prev:
-            //          value == prev_value
-            //      else:
-            //          value == 0
-            let q_memory_not_first = q_memory_not_first(meta);
-            let address_diff = {
-                let address_prev = meta.query_advice(address, Rotation::prev());
-                let address_cur = meta.query_advice(address, Rotation::cur());
-                address_cur - address_prev
-            };
+        ///////////////////////// General constraints /////////////////////////////////
```

Comment:
I noticed that the constraints `0, 1, 2, 4` are missing.  I understand that you'll do them in a future PR.  But could you add a small TODO stating that there are missing general constraints to be implemented?